server/punishments.ts

```diff
@@ -13,6 +13,7 @@
 
 import {FS} from '../lib/fs';
 import {Utils} from '../lib/utils';
+import {getGlobalPunishments} from './chat-plugins/modlog';
```

Comment:
This looks like it would break if you tried to hotpatch only one of chat and punishments.

If you don't want to do a full refactor, you might want to put `getGlobalPunishments` on a global that the plugin can hotpatch.

Reply:
tbh I could just draft this until I refactor it; I don't think the functionality is so vital that it can't wait a few weeks.